pontoon/base/models.py

```diff
@@ -276,36 +276,39 @@ def repository_for_path(self, path):
         else:
             return repo
 
-    def latest_activity(self, locale=None):
+    def latest_activity(self, locale=None, path=None):
         """Get latest activity for project and locale if provided."""
         translations = Translation.objects.filter(entity__resource__project=self)
         if locale:
             translations = translations.filter(locale=locale)
+            if path:
+                translations = translations.filter(entity__resource__path=path)
 
         return translations.latest_activity()
 
-    @property
-    def locales_parts_stats(self):
+    def locales_parts_stats(self, loc=None):
```

Comment:
This method needs to be refactored for performance reasons soon, so I'd rather not touch it at this stage. That being said, it might be that we hit performance issues again (will test on stage), so I'll have to refactor it before merging this PR anyways.